code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -893,9 +893,9 @@
 /datum/action/innate/undeployment/Trigger(trigger_flags)
 	if(!..())
 		return FALSE
-	var/mob/living/silicon/robot/R = owner
+	var/mob/living/silicon/robot/shell_to_disconnect = owner
 
-	R.undeploy()
+	shell_to_disconnect.mainframe.disconnect_shell()
```

Comment:
Why is the unregister signal not in `undeploy` instead?

Other places call `undeploy` directly, so it remains an error on those cases. 

On top of that this has the side effect of sending the ai `"Your remote connection has been reset!"` if they willingly disconnect which might get annoying.